src/base/http/requestparser.cpp

```diff
@@ -30,314 +31,322 @@
 #include "requestparser.h"
 
 #include <QDebug>
-#include <QDir>
+#include <QRegularExpression>
 #include <QStringList>
 #include <QUrl>
 #include <QUrlQuery>
 
-const QByteArray EOL("\r\n");
-const QByteArray EOH("\r\n\r\n");
+#include "base/utils/string.h"
 
-inline QString unquoted(const QString &str)
+using namespace Http;
+using QStringPair = QPair<QString, QString>;
+
+namespace
 {
-    if ((str[0] == '\"') && (str[str.length() - 1] == '\"'))
-        return str.mid(1, str.length() - 2);
+    const QByteArray EOH = QByteArray(CRLF).repeated(2);
 
-    return str;
-}
+    QByteArray chopIfEndsWith(QByteArray in, const QByteArray &str)
+    {
+        if (in.endsWith(str))
+            in.chop(str.size());
+        return in;
+    }
 
-using namespace Http;
+    QList<QByteArray> split(const QByteArray &in, const QByteArray &sep, const QString::SplitBehavior behavior = QString::KeepEmptyParts)
```

Comment:
Shouldn't we have something like QString::splitRef here, to avoid unnecessary data copying? E.g.:
```c++
QList<const QByteArray> splitIntoViews(const QByteArray &in, const QByteArray &sep, const QString::SplitBehavior behavior = QString::KeepEmptyParts)
```

Reply:
I'll try.

>unless I'm wrong, we can process only original data views in all cases until we require to save some of them or convert to another format (e.g. QString).

I imagine that would be possible, however my intention was to write clear code first, performance later. That said, I don't wish to optimize to the extreme in this PR, are you interested to do it later?

Reply:
Nicely. Just do something that doesn't make you too hard. It'll still be better than nothing. I'll add the rest to my to-do list.

Reply:
Done. But there is a limitation in QList: 
>QList's value type must be an assignable data type

That means I cannot use `QList<cosnt QByteArray>` here, thus it's still `QList<QByteArray>`.